ropsten-beacon-chain/config.yaml

```diff
@@ -0,0 +1,66 @@
+# Extends the mainnet preset
+PRESET_BASE: 'mainnet'
+CONFIG_NAME: 'ropstenbeaconchain'
+
+# Genesis
+# ---------------------------------------------------------------
+MIN_GENESIS_ACTIVE_VALIDATOR_COUNT: 100000
+# Monday, May 30th, 2022 3:00:00 PM +UTC
+MIN_GENESIS_TIME: 1653922800
+GENESIS_FORK_VERSION: 0x80000069
+GENESIS_DELAY: 300
```

Comment:
Is this network going to have a permissioned or unpermissioned deposit contract? From [this](https://github.com/eth-clients/merge-testnets/pull/9#issue-1237597250), I'm guessing it's *unpermissioned*:

> (Same deposit contract code as mainnet)

If this is permissionless, I would propose that we extend this `GENESIS_DELAY` significantly for two reasons:

1. It is computationally intensive to generate the genesis state (think seconds to minutes). In a permissionless network, it's impossible to know the genesis state until the `MIN_GENESIS_TIME` has been reached (i.e., a Ropsten execution block includes this timestamp or higher). This is because *all* deposits up until that time will be included in the genesis state. Therefore, with a `GENESIS_DELAY` of 300 secs (5 mins) we're cutting it rather fine for all nodes in the network to generate their states and be ready to go.
2. It's nice for CL clients to be able to cut a release that contains the entire genesis state some days before genesis. This means users can avoid the awkward and little-tested scenario where they need to boot without a known genesis state. We started a testnet in this way for a long time (years) and it's not relevant to testing the merge.

To address my concerns, I propose:

- Set `GENESIS_DELAY: 604800` (7 days, as per [mainnet specs](https://github.com/ethereum/consensus-specs/blob/dev/configs/mainnet.yaml#L32)).
- Set `MIN_GENESIS_TIME: 1653318300` (The current value of `1653922800` minus the new delay of `1653922800` plus the old delay of `300`)

The effect of this is that the first slot of the beacon chain occurs at the same time, but we know what the state is going to be 7 days in advance.

If the network is going to be *permissioned* and we can guarantee that we can make `n` deposits, generate a genesis state and know there will be no more deposits, then there's no real reason for a change.

Reply:
I believe this is the same as prater where there's an initial genesis state in this PR with all the validators already added (no deposits sent for them).  So any deposits to the deposit contract would automatically be post-genesis.  The genesis.ssz is easy to miss in the GitHub diff though because it can't display the binary content.

Reply:
Its a permissionless deposit contract, similar to prater. I do agree with your concerns and I've modified the `config.yaml` as well as updated the `genesis.ssz` to reflect the same. 
`MIN_GENESIS_TIME: 1653318000` with a `GENESIS_DELAY: 604800`. 

I hope the changes are early enough to not affect anyone's releases.

Reply:
If I remember correctly, only Sepolia should be permissioned.

Reply:
Yup, Sepolia will be a permissioned deposit contract. Goerli/Prater + Ropsten will be permissionless.

Reply:
> The genesis.ssz is easy to miss in the GitHub diff though because it can't display the binary content.

Oh good point, I did miss that! If the network is permissionless then we'll have a difference between clients who:

- Use the `genesis.ssz`
- Generate their own genesis state from the deposit contract
    - @terencechain I recall this was Prysm's only behaviour, is that still the case?
    
They could disagree on the initial set of validators and also the exact genesis time.

If we're going to do a true permissionless system, then I think we need to remove the `genesis.ssz` and then add it once we've generated it on/after the `MIN_GENESIS_TIME` (assuming adequate deposits are submitted).

Reply:
Another option would be to:

1. Deploy the deposit contract secretly.
1. Add the 100k validators.
1. Take note of the timestamp of the Ropsten block which includes the 100,000th deposit.
1. Set the `MIN_GENESIS_TIME` to the above timestamp.
1. Set `GENESIS_DELAY` such that genesis happens at the same time as now (31/05/2022).
1. Generate a `genesis.ssz` using some client (LH can do it) and upload it to the repo.

I'm pretty sure that would ensure:

1. We control the initial validator set.
1. Clients who use `genesis.ssz` and who generate their own agree.

This is a little complicated though, perhaps just waiting until 7 days before genesis is the way to go :thinking: 
1. Clients can use the ``

Reply:
All clients must handle the genesis setup used here where validators just magically appear in the genesis.ssz with no deposits because that's how prater works (and all the other recent testnets/devnets I've seen).  You occasionally see prysm users confused because they didn't specify the genesis state - not sure if it's now set by default or not but it can be set.  I was actually considering removing the create-genesis-from-deposits code from Teku because it never gets used anymore.

And if genesis is already known then we don't need a long genesis delay because we already know which deposits are included (none of them - we start just with the validators that magically appeared in the state).

Reply:
> All clients must handle the genesis setup used here where validators just magically appear in the genesis.ssz with no deposits because that's how prater works (and all the other recent testnets/devnets I've seen).

Oh, I didn't realise that we'd broken compatibility with clients who generate their genesis state locally. Goes to show that no one at Lighthouse ever does it anymore. This explains a long term misunderstanding with @protolambda where I could never understand how this method actually worked :sweat_smile: 

If @ajsutton is right about breaking compatibility with generating states manually, then I agree that the changes I suggested are unnecessary. Apologies.

Reply:
Just to recap, I did update the genesis.ssz and config to add a 7 day delay already. I can revert the commit, but i see no harm in including it anyway. We are currently using the same method on prater/all the new devnets(including kiln/kintsugi), hasn't been an issue so far, so I think we wouldn't break compatibility with any clients.  

What do you guys prefer? Should i leave it at the current commit(7 day delay) or revert to the earlier version (5 mins delay)?

Reply:
Seems easier to stick with what's there now rather than changing again.

I've updated Teku with the new config and genesis.  I think the bootnode ENR should stay the same but will let you know if it changes.

Reply:
Great, I'll merge the PR as is then